app/src/main/java/org/mozilla/fenix/settings/address/AddressEditorFragment.kt

```diff
@@ -26,6 +30,15 @@ class AddressEditorFragment : SecureFragment(R.layout.fragment_address_editor) {
 
     private lateinit var addressEditorView: AddressEditorView
     private lateinit var interactor: AddressEditorInteractor
+    private val args by navArgs<AddressEditorFragmentArgs>()
+
+    /**
+     * Returns true if an existing address is being edited, and false otherwise.
+     */
+    private val isEditing: Boolean
+        get() = args.address != null
```

Comment:
I think this is out-of-scope for now, as it we're pretty far along in the implementation of this entire feature. Seeing this got me thinking though: why are needing to pass data between simple screens through navigation arguments?

I think we're intending to model the design of this feature after credit cards. However, I believe the intention moving forward is to follow the architecture defined in the `lib-state` module in A-C. 

If we had created a `Store` that was scoped to this feature (using a `StoreProvider` to share it between fragments), it would have allowed us to avoid this kind of thing as well as the kind of delegate-chaining we see in the controller/interactor. The resulting side-effects in those classes could have been moved instead into Middlewares attached to the Store.

No actions needed, just sharing some thoughts

Reply:
I believe only `AppState` is the only one that persists in memory. So, creating a new `Store` that is shared between fragments won't actually persist the selected Address between fragments. You would still need to pass the selected Address and inform the newly created `Store` of the state, which gets recreated every time you navigate to the fragment. My current mental model is that a `Store` only lives for as long as the fragment does. However, I can totally be wrong about all of this and I have never looked into this deeply enough to confirm these claims - there are people who can provide an answer if we asked in a team channel, which I think would be a nice thing to do.

Reply:
I prefer to use nav args when data is passed from a single fragment to a "child" fragment ( by child fragment in this context I define a fragment that is accessible only from the one fragment) and the data is not really needed in other fragments /places in the app. We could save all things in a store/state only for the one screen, but that would be overkill IMO.

Reply:
Oh you're right, `StoreProvider.get` is fragment-scoped. My thinking is still being influenced some by my last position, here's some details as to why if you're curious:

The last project I worked on was multi-module, which informed some of the architecture. We typically would have a parent fragment/activity as an entry point to a module, which allowed us to create well-defined entry points as well as share data between child fragments in the module. 

For example, imagine having a `Settings` (or even `AutofillSettings`) module that had a `Store` scoped to the lifecycle of a parent fragment.

We could also theoretically add a `StoreProvider.get` method that used the `HomeActivity` to share stores between fragments, but I'd worry about memory bloat. We'd probably need to think through lifecycle retention pretty deeply